python/dgl/graphbolt/internal_utils.py

```diff
@@ -15,6 +17,18 @@
     from setuptools.extern.packaging import version
 
 
+@functools.lru_cache(maxsize=None)
```

Comment:
put it in https://github.com/dmlc/dgl/tree/master/tests/utils?

Reply:
As in, create a new file?

Reply:
We might actually use this functionality because our examples are broken on WSL systems right now because we use in-place pinning by default.

Reply:
Future PR might detech WSL and use out-of-place pinning instead of in-place pinning.